browse/src/views/MonitoringAnalysisView.vue

```diff
@@ -0,0 +1,567 @@
+        ];
+        selectedLabels.value = [];
+      }
+      setFilterTrackTags();
+    });
+
+    watch(
+      [tagTotals, groupTotals, stationTotals, userTotals, datasetFocus],
+      () => {
+        if (!barChart) {
+          return;
+        }
+        if (datasetFocus.value === DataType.Tag) {
+          barChart.data.datasets[0].data = tagTotals.value;
+        } else if (datasetFocus.value === DataType.Group) {
+          barChart.data.datasets[0].data = groupTotals.value;
+        } else if (datasetFocus.value === DataType.Station) {
+          barChart.data.datasets[0].data = stationTotals.value;
+        } else if (datasetFocus.value === DataType.User) {
+          barChart.data.datasets[0].data = userTotals.value;
+        }
+        const nshades = Math.max(
+          Object.keys(barChart.data.datasets[0].data).length,
+          6
+        );
+        barChart.data.datasets[0].backgroundColor = colormap({
+          colormap: "summer",
+          nshades,
+          format: "hex",
+        }).reverse();
+        barChart.update();
+      }
+    );
+
+    watch([trackTags, filterTrackTags], () => {
+      const extractTotalOf =
+        (data: { [key: string]: any }[]) =>
+        (newKey: string, ...children: string[]) => {
+          return _(
+            data.reduce((acc, newCurr) => {
+              const [curr, key] = children.reduce(
+                ([currChild, key], childKey) => {
+                  if (typeof currChild[key] === "object") {
+                    return [currChild[key], childKey];
+                  } else {
+                    return [currChild, key];
+                  }
+                },
+                [newCurr, newKey]
+              );
+              if (curr === null) {
+                return acc;
+              }
+              if (acc.hasOwnProperty(curr[key])) {
+                acc[curr[key]] += 1;
+              } else {
+                acc[curr[key]] = 1;
+              }
+              return acc;
+            }, {})
+          )
+            .toPairs()
+            .orderBy([1], ["desc"])
+            .fromPairs()
+            .value();
+        };
+      const totalOf = extractTotalOf(filterTrackTags.value);
+      tagTotals.value = totalOf("label");
+      stationTotals.value = totalOf("station", "name");
+      groupTotals.value = totalOf("group", "name");
+      userTotals.value = totalOf("labeller");
+    });
+
+    const { tag, station, group, device, user } = route.value.query;
+    selectedLabels.value = tag ? (Array.isArray(tag) ? tag : [tag]) : [];
+    stations.value = station
+      ? Array.isArray(station)
+        ? station
+        : [station]
+      : [];
+    devices.value = device ? (Array.isArray(device) ? device : [device]) : [];
+    groups.value = group ? (Array.isArray(group) ? group : [group]) : [];
+    userFocus.value = user ? (user as string) : null;
+
+    onMounted(async () => {
+      await updateTrackTags();
```

Comment:
If this is potentially a slow query, could be good to have some UI state around it to add a spinner etc.